### DISCUSSION TITLE
Optimize a different airfoil

### DISCUSSION THREAD

**Discussion by hosshvsi** [mdolab/dafoam#646] [Bug]
Dear all,

I was trying to optimize another airfoil (NACA0018) instead of the tutorial case (NACA0012). First, I imported its SS and PS profiles, then changed genFFD.py because the highest point this time was around 0.089. The geometry was created successfully, but when I tried to run the case, this error appeared:

    No coarse levels created, either matrix too small for GAMG or nCellsInCoarsestLevel too large.
    Either choose another solver of reduce nCellsInCoarsestLevel.
   
To address this issue, I tried solving the problem with 12, 4, 2, and 1 cores, but the error remained the same. Additionally, I tried setting nCellsInCoarsestLevel to values 10, 100, and 500, but the error persisted. Finally, I changed the solver in fvSolution from GAMG to PCG like this:

    solver PCG;
    preconditioner DIC;
    tolerance 1e-07;
    relTol 0.05; 
    
However, this time the following error appeared:
       +------------------------------------------------------------------------------+
      pyDAFoam Error: Primal solution failed for the baseline design!            |
     +------------------------------------------------------------------------------+
     Traceback (most recent call last):
      File "/home/dafoamuser/dafoam/packages/miniconda3/lib/python3.8/site-packages/openmdao/core/system.py", line 2927, in 
      _call_user_function
      +------------------------------------------------------------------------------+
       | pyDAFoam Error: 'cruise.coupling.solver' <class DAFoamSolver>: Error       |
       | calling solve_nonlinear(),                      
      +------------------------------------------------------------------------------+
      
Does anyone have any ideas on how to address this issue?
In the attachment you can find initial fvSolution, my new genFFD.py and also PS and SS.
[genFFD.py.txt](https://github.com/user-attachments/files/15792350/genFFD.py.txt)
[NACA0012PS.profile.txt](https://github.com/user-attachments/files/15792355/NACA0012PS.profile.txt)
[NACA0012SS.profile.txt](https://github.com/user-attachments/files/15792356/NACA0012SS.profile.txt)
[fvSolution.txt](https://github.com/user-attachments/files/15792363/fvSolution.txt)

**Comment by friedenhe** [comment]
I would need the full log. The "No coarse levels created" error you saw may be caused by something else.

**Reply by hosshvsi** [reply]
here is the log.output file, as well as the runscript which I used to run this case
[log.output.txt](https://github.com/user-attachments/files/15890992/log.output.txt)
[runScript_v2.py.txt](https://github.com/user-attachments/files/15890993/runScript_v2.py.txt)

**Comment by friedenhe** [comment]
Your FFD points do not fully contain the design surface: check https://dafoam.github.io/mydoc_get_started_faq.html#how-to-fix---warning-xx-points-not-projected-to-tolerance-1e-12-max-error--xxx--rms-error-xxx-list-of-points-is-pt-delta
